requestmanager/server.go

```diff
@@ -49,14 +53,23 @@ func (rm *RequestManager) cleanupInProcessRequests() {
 	}
 }
 
-func (rm *RequestManager) newRequest(p peer.ID, root ipld.Link, selector ipld.Node, extensions []graphsync.ExtensionData) (gsmsg.GraphSyncRequest, chan graphsync.ResponseProgress, chan error) {
+func (rm *RequestManager) newRequest(parentSpan trace.Span, p peer.ID, root ipld.Link, selector ipld.Node, extensions []graphsync.ExtensionData) (gsmsg.GraphSyncRequest, chan graphsync.ResponseProgress, chan error) {
 	requestID := rm.nextRequestID
 	rm.nextRequestID++
 
+	ctx := trace.ContextWithSpan(rm.ctx, parentSpan)
+	var span trace.Span
+	ctx, span = otel.Tracer("graphsync").Start(ctx, "newRequest", trace.WithAttributes(
+		attribute.Int("requestID", int(requestID)),
+	))
+	defer span.End()
```

Comment:
again, I don't the span.End() goes here. It goes somewhere in [terminateRequest](https://github.com/ipfs/go-graphsync/pull/283/files#diff-71edc89e7b7c1e4f58b00139297dc49e2ddb0671d6adfe0d152d569f6ce8e398L165)

Reply:
otherwise you terminate the span once you pass on to the executor

Reply:
unless you have an error with validation or other request setup here, in which case I guess you span.End() there.